tests/js/utils.js

```diff
@@ -190,6 +196,49 @@ export const provideUserInfo = ( registry, extraData = {} ) => {
 	} );
 };
 
+/**
+ * Registers the given module with settings to the given registry.
+ *
+ * @since n.e.x.t
+ * @private
+ *
+ * @param {Object}   registry    Registry object to dispatch to.
+ * @param {Object[]} [extraData] List of module objects to be merged with defaults. Default empty array.
+ */
+export const provideModuleRegistrations = ( registry, extraData = [] ) => {
+	const extraModules = extraData.reduce( ( acc, module ) => {
+		return { ...acc, [ module.slug ]: module };
+	}, {} );
+
+	const moduleIconMap = {
+		adsense: AdsenseIcon,
+		analytics: AnalyticsIcon,
+		optimize: OptimizeIcon,
+		'pagespeed-insights': PagespeedInsightsIcon,
+		'search-console': SearchConsoleIcon,
+		tagmanager: TagManagerIcon,
+	};
+
+	const moduleSlugs = coreModulesFixture.map( ( { slug } ) => slug );
+
+	const modules = coreModulesFixture
+		.concat(
+			extraData.filter( ( { slug } ) => ! moduleSlugs.includes( slug ) ),
+		);
+
+	modules.forEach( ( module ) => {
+		const icon = moduleIconMap[ module.slug ];
+		let moduleData = { ...module, icon };
+		if ( extraModules[ module.slug ] ) {
+			moduleData = { ...moduleData, ...extraModules[ module.slug ] };
+		}
+		delete moduleData.slug;
+		registry.dispatch( CORE_MODULES ).registerModule( module.slug, { ...moduleData } );
+	} );
+
+	registry.dispatch( CORE_MODULES ).receiveGetModules( modules );
```

Comment:
The implementation here is slightly off:
* It should only call `registerModule` for all six modules and not deal with populating server-side module data (`receiveGetModules`).
* Therefore it also shouldn't rely on the `coreModulesFixture`, which is only responsible for server-side module data. Basically, the main part of this function would be to replicate all six `registerModule` calls that in the production code are found within the `assets/js/modules/*/index.js` files into this single utility function, in order to allow us to use the same module registration data in Storybook that the plugin also uses in production.
* In other words, we should probably have for each one of the six modules an entry here with all arguments that are passed to the real `registerModule` call, and combine that base data with `extraData` for that module, if passed, before calling `registerModule`.

Reply:
I made the changes as requested. I had to call the `receiveGetModules` otherwise a number of stories would have to be overwritten. Now I am calling `provideModuleRegistrations` followed by `provideModules` which works as expected.